staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -970,7 +970,8 @@ func (ca *ConnectionAgent) createOrUpdateIfc(attGuestIP, attHostIP gonet.IP, att
 			}
 			ca.LaunchCommand(attNSN, &oldLocalState.LocalNetIfc, oldLocalState.PostDeleteExec, "postDelete", true, false)
 			ca.unsetLocalAttState(attNSN)
-		} else if attHasRemoteIfc {
```

Comment:
Unrelated to immutability: this change makes the code look for and delete stale local and remote interfaces independently as opposed to in mutual exclusion. An attachment should never own more than one interface, but in case there's a bug somewhere in the code this change makes the connection agent self-heal. This change was introduced as a consequence of the problem described in the previous comment.

Reply:
good